src/io.rs

```diff
@@ -24,10 +24,18 @@ mod writer;
 pub(crate) mod der_writer;
 
 /// A serialized positive integer.
-#[derive(Copy, Clone)]
+#[derive(Copy, Clone, Debug)]
 pub struct Positive<'a>(untrusted::Input<'a>);
 
 impl<'a> Positive<'a> {
+    /// Constructs a serialized positive integer from serialized
+    /// untrusted input.
```

Comment:
This specification is not precise enough.

In the test cases you provided, the inputs have a leading zero if it has the highest bit set. This is a DER-specific encoding rule that is used to disambiguate negative and positive values; That rule isn't appropriate for this constructor. Instead, like I mentioned, leading zeros should be disallowed, and all values should be assumed to be positive.

Further, the input must be in big-endian order, and empty inputs are not allowed. Further, leading zeros must be rejected. This means, in particular, that zero is encoded as a single zero byte.

All of these aspects must be documented here, and each of these aspects of the constructor must be tested in integration tests (a test under tests/, e.g. in a file `positive_tests.rs`). In particular, "", 00, 0000, 000000, 0001, 00ff, ff, 00ff00, 01, and some long inputs should be tested (this list isn't intended to be exhaustive).